snuba/datasets/storages/groupedmessages.py

```diff
@@ -38,7 +40,17 @@
     local_table_name="groupedmessage_local",
     dist_table_name="groupedmessage_dist",
     storage_set_key=StorageSetKey.EVENTS,
-    mandatory_conditions=[("record_deleted", "=", 0)],
+    mandatory_conditions=[
+        MandatoryCondition(
+            ("record_deleted", "=", 0),
+            binary_condition(
+                None,
```

Comment:
When does a binary condition have an alias?

Reply:
Technically every expression can have an alias. I think we should remove it from conditions.

Reply:
I wondered if we should have a condition type since there seems to be some differences with generic expressions.

Reply:
That was an option discussed at the beginning of the project.
If the condition type is a subclass of functionCall this is viable and has advantages, but it would not really change much here since beingn a subclass of an expression, conditions would still have an alias.
If they were not subclasses of expressions, then we would have again two ways to represent the same conditions (since clickhouse has conditional functions) and all processing would become more complex